app/models/challenge_friend.rb

```diff
@@ -76,5 +103,46 @@ def self.cancel_challenge(user, challenge_id)
     { message: 'Failed to cancel challenge', status: :unprocessable_entity }
   end
 
+  def self.accept_challenge_request(user, challenge_id)
+    challenge = ChallengeFriend.find_by(id: challenge_id, challengee: user, status: 'pending')
+
+    return { message: 'Challenge not found or not eligible for acceptance', status: :not_found } if challenge.nil?
+
+    if user.coins < challenge.amount_of_betting_coin
+      return { message: 'Insufficient coins to accept this challenge', status: :unprocessable_entity }
+    end
+
+    ActiveRecord::Base.transaction do
+      user.update!(coins: user.coins - challenge.amount_of_betting_coin)
+      challenge.update!(status: 'accepted')
```

Comment:
try to use .increament! and .decrement method

Reply:
increment_coins and decrement_coins added in user model.